prac_07/project_management.py

```diff
@@ -0,0 +1,129 @@
+    date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+    projects.sort(key=attrgetter('start_date'))
+    for project in projects:
+        if project.start_date > date:
+            print(project)
+
+
+def save_to_file(projects):
+    file_name = input("Filename: ")
+    with open(file_name, 'w') as in_file:
+        print("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage", file=in_file)
+        for project in projects:
+            print(project, file=in_file)
+            # Could not figure out a way to write this to a file in the original formatting
+
+
+def update_projects(projects):
+    for i, project in enumerate(projects):
+        i += 0
+        print(i, project)
+    project_to_modify = get_valid_number("Project choice: ", projects)
+    print(projects[project_to_modify])
+    new_percentage = int(input("New Percentage: "))
+    if new_percentage != "":
+        projects[project_to_modify].completion_percentage = new_percentage
+    new_priority = int(input("New Priority: "))
+    if new_priority != "":
+        projects[project_to_modify].priority = new_priority
+    return projects
+
+
+def get_valid_number(user_input, projects):
+    """Gets a valid number and returns it"""
+    valid_number = False
+    while not valid_number:
+        try:
+            number = int(input(user_input))
+            if len(projects) <= number <= -1:
+                print("Number must be within numbers listed above")
+            else:
+                valid_number = True
+                return number
+        except ValueError:
+            print("Invalid input; enter a valid number")
+
+
+def add_new_project(projects):
+    print("Lets add a new project: ")
+    name = input("Name: ")
+    date_string = input("Start date (dd/mm/yy): ")
+    date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+    priority = input("Priority: ")
+    cost = float(input("Cost Estimate: $"))
+    completion = int(input("Percent completed: "))
+    project_to_add = Project(name, date, priority, cost, completion)
+    projects.append(project_to_add)
+    return projects
+
+
+def display_projects(projects):
+    print("Incomplete Projects:")
+    projects.sort(key=attrgetter('priority'))
+    for project in projects:
+        if project.completion_percentage < 100:
+            print(project)
+    print("Completed Projects:")
+    for project in projects:
+        if project.completion_percentage == 100:
+            print(project)
+
+
+def load_projects_from_file(projects):
+    file_name = input("Filename: ")
+    with open(file_name, 'r') as in_file:
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split('\t')
+            date_string = parts[1]
+            date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+            project = Project(parts[0], date, parts[2], parts[3], int(parts[4]))
+            projects.append(project)
+    return projects
+
+
+main()
```

Comment:
This was a pretty big task, good job on getting it done. 
Code looks really good 👍